auth_server/src/realms.rs

```diff
@@ -61,6 +65,39 @@ pub async fn receive_realm_pings(auth_db: std::sync::Arc<AuthDatabase>) -> Resul
     }
 }
 
+async fn get_realm_list(auth_database: std::sync::Arc<AuthDatabase>, account_id: u32) -> Result<Vec<Realm>> {
+    //TODO(wmxd): it will be good idea to cache the database stuff
+    //TODO(wmxd): for now it will be better select realms and number_of_chars in one database trip (eg: left join)
```

Comment:
Going to do this before merging, should be simple enough

Reply:
It doesn't seem like SQLX is very good at handling `joins` in a nice re-usable way. If you have a look at what I've done in d3be028f7b45e2259c6f0d0483369301c6544fa1, you'll see that I had to duplicate `DBRealm` and add the `num_characters` column as an `Option` for the left join. Super annoying.

I thought about just having the `Option` in the original DBRealm and adding a literal like `NULL as num_characters` or `0 as num_characters` to the normal `get_all_realms` query, but it doesn't seem like there's a good way of defining a nullable literal in an sql query. Maybe the answer is just to use `query!` for joins and keep them as anonymous records, but then they can't be returned from functions.

Reply:
Having both `DBRealm` and `DBRealmWithNumCharacters` is totally fine by me.